std_traffic/utils/data_utils.py

```diff
@@ -0,0 +1,166 @@
+"""Utility functions to process data"""
+
+from os import environ
+import logging
+
+import psycopg2
+import psycopg2.extras
+from psycopg2 import sql
+from psycopg2.extensions import AsIs
+from psycopg2.extensions import connection
+
+import pandas as pd
+import numpy as np
+
+
+log = logging.getLogger(__name__)
+
+
+def get_db(
+        database: str,
+        user: str = None,
+        password: str = None,
+        host: str = None,
+        port: int = None,
+) -> connection:
+    """Create a connection to a database.
+    The default connection parameters are taken from ENV
+    """
+
+    if not user:
+        user = environ['DB_USER']
+    if not password:
+        password = environ['DB_PASSWORD']
+    if not host:
```

Comment:
Couldn't it go directly in the default parameter definition?

Reply:
@RuggeroFabbiano If you mean

```
def foo(arg_1, arg_2=environ[KEY_2])
```

that would work the same, but it would force the evaluation of `environ[KEY_2]` much before the function is used. It would be evaluated whenever the whole module `data_utils.py` is imported, and that feels unnatural to me.
For that reason, the import would fail with a `KeyError` if `KEY_2` doesn't exist in `environ`, even if the user meant to not use the default value for the argument.
This last problem could be solved using `getenv(KEY_2)` instead of 'envirion[...]` but it still looks odd to me in the function definition.
Is this what you meant?

Reply:
Yes it is, and I see it now.